src/library/autodj/audioscrobbler.cpp

```diff
@@ -0,0 +1,278 @@
+#include <QNetworkReply>
+#include <QDomDocument>
+#include <QList>
+#include <QByteArray>
+#include <QApplication>
+
+#include "audioscrobbler.h"
+#include "library/dao/autodjcratesdao.h"
+
+/* Large parts of this code are copied from the https://sayonara-player.com/ project */
```

Comment:
// The concept is adopted from the the https://sayonara-player.com/ project

Reply:
Are you in contact with Lucio Carreras? I think it cant hurt to say hello and thank you for the good ideas.

Reply:
Yes, I sent him a hello message with a link to this discussion.

Reply:
Hi all, got a nice answer from Lucio Carreras: 

"... Just mention me and Sayonara in your copyright file and in the source file if you copy the code. If you only take the code as inspiration you can add a "Inspired by... " in the header of the source file and in the copyright file. No need to be formal though, just write it as you want it to write.

I am happy if I could help you

Lucio"